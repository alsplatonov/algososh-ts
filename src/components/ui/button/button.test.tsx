import React from 'react';
import { render } from '@testing-library/react';
import { Button } from "./button";

describe('Button Component', () => {
  it('renders a button with text', () => {
    const { tree } = render(<Button text="Click me" />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders a button without text', () => {
    const { tree } = render(<Button />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders a disabled button', () => {
    const { tree } = render(<Button text="Disabled" disabled />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders a button with a loading indicator', () => {
    const { asFragment } = render(<Button isLoader />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('calls the onClick callback when clicked', () => {
    const onClickMock = jest.fn();
    const { asFragment, getByText } = render(
      <Button text="Click me" onClick={onClickMock} />
    );
    const button = getByText('Click me');

    button.click();
    expect(onClickMock).toHaveBeenCalledTimes(1);

    expect(asFragment()).toMatchSnapshot();
  });
});

