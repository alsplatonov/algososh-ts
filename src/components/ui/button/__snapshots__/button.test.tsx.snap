// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component calls the onClick callback when clicked 1`] = `
<DocumentFragment>
  <button
    class="text text_type_button text_color_primary button undefined false "
    type="button"
  >
    <p
      class="text undefined"
    >
      Click me
    </p>
  </button>
</DocumentFragment>
`;

exports[`Button Component renders a button with a loading indicator 1`] = `
<DocumentFragment>
  <button
    class="text text_type_button text_color_primary button undefined loader "
    disabled=""
    type="button"
  >
    <img
      alt="Загрузка."
      class="loader_icon"
      src="loader.svg"
    />
  </button>
</DocumentFragment>
`;

exports[`Button Component renders a button with text 1`] = `
<DocumentFragment>
  <button
    class="text text_type_button text_color_primary button undefined false "
    type="button"
  >
    <p
      class="text undefined"
    >
      Click me
    </p>
  </button>
</DocumentFragment>
`;

exports[`Button Component renders a button without text 1`] = `
<DocumentFragment>
  <button
    class="text text_type_button text_color_primary button undefined false "
    type="button"
  >
    <p
      class="text undefined"
    />
  </button>
</DocumentFragment>
`;

exports[`Button Component renders a disabled button 1`] = `
<DocumentFragment>
  <button
    class="text text_type_button text_color_primary button undefined false "
    disabled=""
    type="button"
  >
    <p
      class="text undefined"
    >
      Disabled
    </p>
  </button>
</DocumentFragment>
`;
